wellcomecollection/wellcomecollection.org
# Remove e2e tests not required label

## What does this change?

Part of https://github.com/wellcomecollection/platform/issues/5740
We're setting up alternatives to weco-bot everywhere it's used because it doesn't conform the the SSO standard we're set. Weco-bot is a "fake" team member that has its own credentials, used to perform automated tasks.

Before this change, one of these tasks was to call Github API from a Buildkite pipeline run, in order to add the "e2e not required" label to the PR. This label was only used to "remember" on later commits if the developer had asked that e2e tests never be run on the PR.

In the absence of this "e2e not required" label, Buildkite will ask if the dev wants to run the e2e tests or not every time a commit is built. 

As agreed with the experience team, we're removing the option to add the "e2e tests not required" label on PRs from Buildkite. 
After this change, every time a commit is pushed to a branch and triggers a build & test in Buildkite, the dev will need to select whether or not they want e2e tests run in order to complete the pipeline.

## How to test

Checkout a new branch, make any trivial change and push to the remote. In https://buildkite.com/wellcomecollection/wc-dot-org-build-plus-test you should see you commit building, with a step asking you whether you want e2e tests or not. 
Select Yes and see that the tests run and the pipeline completes.
Push another commit to the branch, select No and see that the pipeline completes without running the e2e tests

## How can we measure success?

Weco-bot is not involved in running/managing e2e tests on PR branches

## Have we considered potential risks?

The dev still has to make the deliberate decision to NOT run the e2e tests for the pipeline to complete
